app/src/main/java/com/fcaputo/parkingapp/mvp/view/ReservationView.kt

```diff
@@ -0,0 +1,157 @@
+            .build()
+
+        datePicker.addOnPositiveButtonClickListener {
+            val dateString = DateTimeCustomFormat.formatDate(Date(it))
+            val editText = ViewBindings.findChildViewById<TextInputEditText>(binding.root, editTextId)
+            editText?.setText(dateString)
+        }
+
+        datePicker.show((activity as AppCompatActivity).supportFragmentManager, activity?.getString(R.string.date_picker_tag))
+    }
+
+    override fun onTimeInputPressed(onInputClick: (CharSequence?, Int) -> Unit) {
+        for (tInput in listOf(binding.etStartTime, binding.etEndTime)) {
+            tInput.setOnClickListener {
+                onInputClick((it as TextInputEditText).text, it.id)
+            }
+        }
+    }
+
+    override fun showTimePicker(timeSelected: LocalTime, editTextId: Int) {
+        val timePicker = MaterialTimePicker.Builder()
+            .setTimeFormat(TimeFormat.CLOCK_12H)
+            .setHour(timeSelected.hour)
+            .setMinute(timeSelected.minute)
+            //.setTitleText("Select reservation time")
+            .build()
+
+        timePicker.addOnPositiveButtonClickListener {
+            val timeString = DateTimeCustomFormat.formatTime(timePicker.hour, timePicker.minute)
+            val editText = ViewBindings.findChildViewById<TextInputEditText>(binding.root, editTextId)
+            editText?.setText(timeString)
+        }
+
+        timePicker.show((activity as AppCompatActivity).supportFragmentManager, activity?.getString(R.string.time_picker_tag))
+    }
+
+    override fun getTextFieldsContents(): List<CharSequence?> {
+        return listOf(
+            binding.etStartDate,
+            binding.etStartTime,
+            binding.etEndDate,
+            binding.etEndTime,
+            binding.etParkingLot,
+            binding.etSecurityCode
+        ).map{ it.text }
+    }
+
+    override fun getStartDate(): String = getTextFromEditText(binding.etStartDate)
+
+    override fun getStartTime(): String = getTextFromEditText(binding.etStartTime)
+
+    override fun getEndDate(): String = getTextFromEditText(binding.etEndDate)
+
+    override fun getEndTime(): String = getTextFromEditText(binding.etEndTime)
+
+    override fun getParkingLotNumber(): String = getTextFromEditText(binding.etParkingLot)
+
+    override fun getSecurityCode(): String = getTextFromEditText(binding.etSecurityCode)
+
+    private fun getTextFromEditText(editText: TextInputEditText) = (editText.text as CharSequence).toString()
+
+    override fun showErrorMessage(error: ValidationErrorType) {
+        val (titleId, messageId) = getTitleMessageFromError(error)
+
+        MaterialAlertDialogBuilder(activity as AppCompatActivity)
+            .setTitle(titleId)
+            .setMessage(messageId)
+            .setPositiveButton(R.string.error_dialog_got_it_text, null)
+            .show()
+    }
+
+    private fun getTitleMessageFromError(error: ValidationErrorType): Pair<Int, Int> {
+        return when(error){
```

Comment:
this logic should be in the presenter, not here. This is too complex for a view

Reply:
How can this be done in the presenter? it would need to be aware of the resource strings to perform the translation